### ISSUE TITLE
How to stop the process?

### ISSUE THREAD

**Issue by HuakunShen** [DoumanAsh/clipboard-master#8]
I am running it using a thread.

```rust
thread::spawn(async move {
      let _ = Master::new(Handler).run();
});
```

But don't know how to stop it. Maybe a `flag` can be passed in to stop the thread?

**Comment by HuakunShen** [comment]
Something like https://users.rust-lang.org/t/using-arc-to-terminate-a-thread/81533 or https://stackoverflow.com/questions/26199926/how-to-terminate-or-suspend-a-rust-thread-from-another-thread

**Comment by DoumanAsh** [comment]
Right now there is no way to stop it because I just never had need.
I can consider implement some signal in a simple manner

It also can be problematic because thread blocks waiting for clipboard event (windows and X11)

**Comment by DoumanAsh** [comment]
@HuakunShen btw what platform you use?

**Comment by HuakunShen** [comment]
@DoumanAsh 
> @HuakunShen btw what platform you use?

I am using MacOS but I am actually building cross platform apps for all 3 OS.

Thank you for your work!

**Comment by DoumanAsh** [comment]
@HuakunShen You can try my branch on mac os, but x11 needs more thinking on implementing shutdown as X11 bindings in Rust are quite crappy for use

**Comment by HuakunShen** [comment]
It works on Mac, thanks. Any update to X11? I read the source code. The X11 version is indeed so complicated.

**Comment by DoumanAsh** [comment]
No, sorry, unfortunately I didn't find any way to do it properly with X11, there might be some ways, but I'm yet to invest time to investigate it

**Comment by HuakunShen** [comment]
Thanks! I will try to get this and Wayland working when I got time.

**Comment by DoumanAsh** [comment]
I'll appreciate PR, but please note that most likely you'll have to re-write whole X11 code base 
`x11-clipboard` doesn't expose any functionality to perform timed await for change so likely you need to use `x11` crate directly

**Comment by DoumanAsh** [comment]
@HuakunShen I published `4.0.0-beta.3`  that has significant refactoring to x11 clipboard and introduces Shutdown channel for all platforms

**Comment by HuakunShen** [comment]
@DoumanAsh Thank you so much!
